src/user-data.d.ts

```diff
@@ -32,12 +32,14 @@ type FirestoreSemester = {
   readonly courses: readonly FirestoreSemesterCourse[];
 };
 
-type FirestoreCollegeOrMajorOrMinor = { readonly acronym: string };
 type FirestoreAPIBExam = {
   readonly type: 'AP' | 'IB';
   readonly score: number;
   readonly subject: string;
+  readonly optIn?: Record<string, string[]>;
```

Comment:
Two nits:
- Make the record and array readonly to be consistent with the rest of the datatypes
- Name the record key to make it self-documenting

So something like:
```ts
type APIBOptInOut = {
  // Values are slot names
  readonly [requirementName: string]: readonly string[];
};
```